gdal/alg/viewshed.cpp

```diff
@@ -340,38 +333,47 @@ CPLErr GDALViewshedGenerate(GDALRasterBandH hBand, const char* pszTargetRasterNa
     int iPixel, iLine;
     double dfDistance2 = dfMaxDistance * dfMaxDistance;
 
+    /* If we can't get a SemiMajor axis from the SRS, it will be
+     * SRS_WGS84_SEMIMAJOR
+    */
+    OGRErr eSphereError;
+    double dfSphereDiameter = OSRGetSemiMajor(GDALGetSpatialRef(hDstDS.get()), &eSphereError ) * 2.0;
```

Comment:
use poSRS = dstDS->GetSpatialRef() , and test it against null,  and poSRS->GetSemiMajor()

Reply:
I guess that if we have a null SRS or GetSemiMajor() fail, we could just set an infinite diameter to have a flat Earth based computation (a CPLDebug() could be appropriate though). The computation that takes into account the diameter should work fine with std::numeric_limits<double>::infinity()

Reply:
handled in 81afe4574c7726b367b54c66e81b0b850e35e5ad